implementations/rust/ockam/ockam/src/node.rs

```diff
@@ -0,0 +1,329 @@
+            .import_private_identity(identity_history, secret)
+            .await
+    }
+
+    /// Import an Identity given that was exported as a hex-encoded string
+    pub async fn import_identity_hex(&self, data: &str) -> Result<Identity> {
+        self.identities_creation().import_identity_hex(data).await
+    }
+
+    /// Spawns a SecureChannel listener at given `Address` with given [`SecureChannelListenerOptions`]
+    pub async fn create_secure_channel_listener(
+        &self,
+        identity: &Identity,
+        address: impl Into<Address>,
+        options: impl Into<SecureChannelListenerOptions>,
+    ) -> Result<()> {
+        self.secure_channels()
+            .create_secure_channel_listener(&self.context, identity, address, options)
+            .await
+    }
+
+    /// Initiate a SecureChannel using `Route` to the SecureChannel listener and [`SecureChannelOptions`]
+    pub async fn create_secure_channel(
+        &self,
+        identity: &Identity,
+        route: impl Into<Route>,
+        options: impl Into<SecureChannelOptions>,
+    ) -> Result<Address> {
+        self.secure_channels()
+            .create_secure_channel(&self.context, identity, route, options)
+            .await
+    }
+
+    /// Extended function to create a SecureChannel with [`SecureChannelOptions`]
+    pub async fn create_secure_channel_extended(
+        &self,
+        identity: &Identity,
+        route: impl Into<Route>,
+        options: impl Into<SecureChannelOptions>,
+        timeout: Duration,
+    ) -> Result<Address> {
+        self.secure_channels()
+            .create_secure_channel_extended(&self.context, identity, route, options, timeout)
+            .await
+    }
+
+    /// Start a new worker instance at the given address
+    pub async fn start_worker<NM, NW>(
+        &self,
+        address: impl Into<Address>,
+        worker: NW,
+        incoming: impl IncomingAccessControl,
+        outgoing: impl OutgoingAccessControl,
+    ) -> Result<()>
+    where
+        NM: Message + Send + 'static,
+        NW: Worker<Context = Context, Message = NM>,
+    {
+        self.context
+            .start_worker(address, worker, incoming, outgoing)
+            .await
+    }
+
+    /// Start a new processor instance at the given address
+    pub async fn start_processor<P>(
+        &self,
+        address: impl Into<Address>,
+        processor: P,
+        incoming: impl IncomingAccessControl,
+        outgoing: impl OutgoingAccessControl,
+    ) -> Result<()>
+    where
+        P: Processor<Context = Context>,
+    {
+        self.context
+            .start_processor(address, processor, incoming, outgoing)
+            .await
+    }
+
+    /// Signal to the local runtime to shut down
+    pub async fn stop(&mut self) -> Result<()> {
+        self.context.stop().await
+    }
+
+    /// Send a message to an address or via a fully-qualified route
+    pub async fn send<R, M>(&self, route: R, msg: M) -> Result<()>
```

Comment:
I think some duplication of API to this struct makes sense, but sending and receiving don't make sense to me. Those things a related very tight with an `Address` assigned to the specific `Context` instance, I'm concerned that this relation is now kinda of lost. `send_and_receive` though, creates a new detached `Context`, so it's fine.

Reply:
Let's discuss this further because I think that we should rework the notion of `Context`.